src/garage/torch/modules/mlp_module.py

```diff
@@ -56,39 +54,87 @@ def __init__(self,
-            self._layers.append(layer)
+            hidden_layers = nn.Sequential()
+            if layer_normalization:
+                hidden_layers.add_module('layer_normalization',
+                                         nn.LayerNorm(prev_size))
+            linear_layer = nn.Linear(prev_size, size)
+            hidden_w_init(linear_layer.weight)
+            hidden_b_init(linear_layer.bias)
+            hidden_layers.add_module('linear', linear_layer)
+
+            if hidden_nonlinearity:
+                hidden_layers.add_module('non_linearity',
+                                         _NonLinearity(hidden_nonlinearity))
+
+            self._layers.append(hidden_layers)
             prev_size = size
 
-        layer = nn.Linear(prev_size, output_dim)
-        output_w_init(layer.weight)
-        output_b_init(layer.bias)
-        self._layers.append(layer)
+        output_layer = nn.Sequential()
+        linear_layer = nn.Linear(prev_size, output_dim)
+        output_w_init(linear_layer.weight)
+        output_b_init(linear_layer.bias)
+        output_layer.add_module('linear', linear_layer)
 
-    def forward(self, input_val):
-        """Forward method."""
-        x = input_val
-        for layer in self._layers[:-1]:
-            x = layer(x)
-            if self._hidden_nonlinearity is not None:
-                x = self._hidden_nonlinearity(x)
-            if self._layer_normalization:
-                x = nn.LayerNorm(x.shape[1])(x)
+        if output_nonlinearity:
+            output_layer.add_module('non_linearity',
+                                    _NonLinearity(output_nonlinearity))
+        self._layers.append(output_layer)
+
+    # pylint: disable=arguments-differ
+    def forward(self, input_value):
+        """Forward method.
+
+        Args:
+            input_value (torch.Tensor): input values
 
-        x = self._layers[-1](x)
-        if self._output_nonlinearity is not None:
-            x = self._output_nonlinearity(x)
+        Returns:
+            torch.Tensor: output value
 
+        """
+        x = input_value
+        for layer in self._layers:
+            x = layer(x)
         return x
+
+
+class _NonLinearity(nn.Module):
+    """Wrapper class for non linear function or module.
+
+    Args:
+        non_linear (callable or type): non linear function or type to be
+            wrapped.
+
+    """
+
+    def __init__(self, non_linear):
+        super().__init__()
+
+        if isinstance(non_linear, type):
+            self.module = non_linear()
+        elif callable(non_linear):
+            self.module = copy.deepcopy(non_linear)
+        else:
+            raise ValueError(
+                'Non linear function {} is not supported'.format(non_linear))
+
+    # pylint: disable=arguments-differ
+    def forward(self, input_value):
+        """Forward method.
+
+        Args:
+            input_value (torch.Tensor):  input values
+
+        Returns:
+            torch.Tensor:output value
```

Comment:
spacing here and in args is weird. please fix.

Argument descriptions should capitalize the first letter, just like an english sentence.

Reply:
fixed